virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -175,106 +288,343 @@ def update(self, time_index: int) -> None:
             * self.data["leaf_area_index"].sum(dim="layers")
         )
 
-        # Mean soil moisture profile, [] and Runoff, [mm]
-        soil_moisture_only, surface_runoff = calculate_soil_moisture(
+        # Calculate soil moisture, [relative water content], vertical flow, [m3/time
+        # step], soil evaporation, [mm], and surface runoff, [mm]
+        soil_hydrology = calculate_soil_moisture(
             layer_roles=self.layer_roles,
+            layer_heights=self.data["layer_heights"],
             precipitation_surface=precipitation_surface,
             current_soil_moisture=self.data["soil_moisture"],
+            temperature=self.data["air_temperature"]
+            .isel(layers=len(self.layer_roles) - self.layer_roles.count("soil") - 2)
+            .drop_vars(["layer_roles", "layers"]),
+            relative_humidity=self.data["relative_humidity"]
+            .isel(layers=len(self.layer_roles) - self.layer_roles.count("soil") - 2)
```

Comment:
As `len(self.layer_roles) - self.layer_roles.count("soil") - 2` is found twice, I think this is a good candidate to be calculated earlier and then just supplied to `.isel(layers=)`

Reply:
Good point. I decided to select the index by name as there is only one `subcanopy` layer. This only needs changing if the layer roles structure is changed which I think is unlikely.